docker/docker-bake-rust-all.hcl

```diff
@@ -0,0 +1,142 @@
+  default = "${AWS_ECR_ACCOUNT_NUM}.dkr.ecr.us-west-2.amazonaws.com/aptos"
+}
+
+variable "gh_image_cache" {
+  default = "ghcr.io/aptos-labs/aptos-core"
+}
+
+# images with IMAGE_TARGET=release for rust build
+group "release" {
+  targets = [
+    "validator",
+    "indexer",
+    "safety-rules",
+    "tools",
+    "init",
+    "txn-emitter",
+  ]
+}
+
+# images with IMAGE_TARGET=test for rust build
+group "test" {
+  targets = [
+    "faucet",
+    "forge",
+  ]
+}
+
+target "_common" {
+  dockerfile = "docker/rust-all.Dockerfile"
+  context    = "."
+  cache-from = [
+    // need to repeat all images here until https://github.com/docker/buildx/issues/934 is resolved
+    generate_cache_from("validator"),
+    generate_cache_from("indexer"),
+    generate_cache_from("validator_tcb"),
+    generate_cache_from("tools"),
+    generate_cache_from("init"),
+    generate_cache_from("txn-emitter"),
+    generate_cache_from("faucet"),
+    generate_cache_from("forge"),
+  ]
+  labels = {
+    "org.label-schema.schema-version" = "1.0",
+    "org.label-schema.build-date"     = "${BUILD_DATE}"
+    "org.label-schema.vcs-ref"        = "${GIT_REV}"
+  }
+  args = {
+    IMAGE_TARGET = "release"
+  }
+}
+
+target "validator" {
+  inherits = ["_common"]
+  target   = "validator"
+  cache-to = generate_cache_to("validator")
+  tags     = generate_tags("validator")
+}
+
+target "indexer" {
+  inherits = ["_common"]
+  target   = "indexer"
+  cache-to = generate_cache_to("indexer")
+  tags     = generate_tags("indexer")
+}
+
+target "safety-rules" {
+  inherits = ["_common"]
+  target   = "safety-rules"
+  cache-to = generate_cache_to("validator_tcb")
+  tags     = generate_tags("validator_tcb")
+}
+
+target "tools" {
+  inherits = ["_common"]
+  target   = "tools"
+  cache-to = generate_cache_to("tools")
+  tags     = generate_tags("tools")
+}
+
+target "init" {
+  inherits = ["_common"]
+  target   = "init"
+  cache-to = generate_cache_to("init")
+  tags     = generate_tags("init")
+}
+
+target "txn-emitter" {
+  inherits = ["_common"]
+  target   = "txn-emitter"
+  cache-to = generate_cache_to("txn-emitter")
+  tags     = generate_tags("txn-emitter")
+}
+
+target "faucet" {
+  inherits = ["_common"]
+  target   = "faucet"
+  cache-to = generate_cache_to("faucet")
+  tags     = generate_tags("faucet")
+  args = {
+    IMAGE_TARGET = "test"
+  }
+}
+
+target "forge" {
+  inherits = ["_common"]
+  target   = "forge"
+  cache-to = generate_cache_to("forge")
+  tags     = generate_tags("forge")
+  args = {
+    IMAGE_TARGET = "test"
+  }
+}
+
+function "generate_cache_from" {
+  params = [target]
+  result = "type=registry,ref=${gh_image_cache}/${target}"
+}
+
+function "generate_cache_to" {
+  params = [target]
+  result = ["type=registry,ref=${gh_image_cache}/${target},mode=max"]
+}
+
+function "generate_tags" {
+  params = [target]
+  result = ["${ecr_base}/${target}:dev_${GIT_REV}", "${ecr_base}/${target}:${GIT_REV}", "${ecr_base}/${target}:${GIT_SHA1}"]
```

Comment:
can we tag all of these with a prefix? We're using prefixes to clean up old images, hence `dev_<REV>` and `<BRANCH>_<REV>` today

Reply:
how does that cleanup work? Happy to have a chat about it.
In my opinion it's the simplest to always just tag everything the full git_sha1 as canonical identifier and not use any other tags, especially since they're kinda moving targets. But I can see how pruning images may become slightly easier with branch prefixed tags.